### PR TITLE
Configuring dagger into the project to use it as the dependency injection library

### PR DESCRIPTION
**Pull request by djeaneGlobant**

### CODE REVIEW

File: app/src/main/res/layout/activity_main.xml [hunk 1]
```diff
@@ -6,10 +6,11 @@
     android:layout_height="match_parent"
     tools:context=".MainActivity">
 
-    <TextView
+    <Button
+        android:id="@+id/button"
         android:layout_width="wrap_content"
         android:layout_height="wrap_content"
-        android:text="Hello World!"
+        android:text="Ir para segunda activity!"
```
Comment (jpaezz) [inline app/src/main/res/layout/activity_main.xml]: should'nt we be using english in all our texts labels?

File: gradle.properties [hunk 2]
```diff
@@ -7,7 +7,7 @@
 # Specifies the JVM arguments used for the daemon process.
 # The setting is particularly useful for tweaking memory settings.
 org.gradle.jvmargs=-Xmx2048m -Dfile.encoding=UTF-8
-# When configured, Gradle will run in incubating parallel mode.
+# When configurned, Gradle will run in incubating parallel mode.
```
Comment (jpaezz) [inline gradle.properties]: Strange typo in this comment

File: app/src/main/res/layout/activity_main.xml (hunk 1 above)
Comment (djeaneGlobant) [reply app/src/main/res/layout/activity_main.xml]: Yes! That was for test. I will make an adjustment 👍

File: gradle.properties (hunk 2 above)
Comment (djeaneGlobant) [reply gradle.properties]: Done! 👍

File: app/src/main/res/layout/activity_main.xml (hunk 1 above)
Comment (djeaneGlobant) [reply app/src/main/res/layout/activity_main.xml]: Done! 👍

File: moviesexample/src/main/java/com/example/movies/domain/PopularMoviesUseCase.kt [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+
+interface PopularMoviesUseCase {
+    operator fun invoke()
+}
+
+class PopularMoviesUseCaseImpl @Inject constructor(
+    private val repository: @JvmSuppressWildcards PopularMoviesRepository
+): PopularMoviesUseCase {
+
+    override fun invoke(){
```
Comment (juans-nino) [inline moviesexample/src/main/java/com/example/movies/domain/PopularMoviesUseCase.kt:14]: add a space before {

File: network/src/main/java/com/example/networkmodule/api/MoviesApi.kt [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+
+import com.google.gson.annotations.SerializedName
+import io.reactivex.Single
+import retrofit2.http.GET
+import retrofit2.http.Query
+
+interface MoviesApi {
+
+    @GET("popular")
+    fun getPopularMovies(@Query("api_key") apiKey: String): MoviesResponse
```
Comment (juans-nino) [inline network/src/main/java/com/example/networkmodule/api/MoviesApi.kt:11]: We can create a local constant for "api_key"

File: moviesexample/src/main/java/com/example/movies/viewmodel/PopularMoviesViewModel.kt [hunk 5]
```diff
@@ -0,0 +1,14 @@
...
+import javax.inject.Inject
+
+class PopularMoviesViewModel @Inject constructor(
+    private val getPopularMovies: PopularMoviesUseCase
+): ViewModel(){
+
+    fun getMovies(){
+        getPopularMovies.invoke()
+    }
+}
```
Comment (juans-nino) [inline moviesexample/src/main/java/com/example/movies/viewmodel/PopularMoviesViewModel.kt:14]: Review- add space format before {

**Review by jpaezz — APPROVED** [review]
Me and Juan can fix what's left.

**Review by jpaezz — APPROVED** [review]

**Review by jpaezz — APPROVED** [review]
-

**Review by juans-nino — APPROVED** [review]

**Pull request merged**